api/src/main/java/org/openmrs/module/idgen/SequentialIdentifierGenerator.java

```diff
@@ -31,10 +31,11 @@
 	private Integer minLength; // If > 0, will always return identifiers with a minimum of this length
 	private Integer maxLength; // If > 0, will always return identifiers no longer than this length
     private String baseCharacterSet; // Enables configuration in appropriate Base
-
+    private String prefixProviderBean;
```

Comment:
I wonder if it would be a bad idea to just overload the existing "prefix" property and how it is used by the SequentialIdentifierGenerator.  For example, interpret a "prefix" value of "locationBasedPrefixProvider:Location Code" as "generate the prefix using bean:configuration" settings.  If the value does not resolve to a value bean:configuration combination, then interpret it as a literal prefix.  Thoughts - this would seem to solve 2 problems - 1. we don't need to add any new columns to the data model, and 2. we don't need to hard-code any specific location attribute types in the module.

Reply:
So you would use the prefix property as a way to convey directions about how to generate prefixes? It could always potentially be interpreted as
>prefix provider bean name -colon- some other instructions

If the bean name is that of `LocationBasedPrefixProvider` then what comes after the colon is the name of the location attribute type required by the algorithm.

If the bean name doesn't point to any known bean, then the whole string should be treated as an actual constant prefix.

Did I get that right?

Reply:
Yes, that's what I was saying.  I'm not 100% sure it's a better idea, but wanted to throw it out there...

Reply:
The downside is that there is an implementation out there that will never be able to prefix their identifiers with `locationBasedPrefixProvider:*`... ;-)

I like and dislike it at the same time. I like the inexpensive way to not touch the data model, clearly. And I dislike the fact that it's quite hacky.
I mean on one end we agree to bring a new logic into `SequentialIdentifierGenerator` but then we make it quite convoluted to configure the behaviour of that new logic.

Reply:
My thought was that we change the _intent_ behind "prefix" to mean "prefixConfiguration".  We can't easily change the name due to backwards-compatibility.  And this is fully backwards-compatible because a static value will be interpreted as configuring a prefix with this static value.  I could also go either way.